pg_search/src/postgres/vacuum.rs

```diff
@@ -49,6 +49,29 @@ pub extern "C" fn amvacuumcleanup(
         })
         .unwrap_or_else(|err| panic!("error loading index writer from directory: {err}"));
 
+    // Merge segments
+    let mut futures = vec![];
+    let merge_policy = writer.get_merge_policy();
+    let segments = search_index
+        .load_metas()
+        .expect("should be able to load metas")
+        .segments;
+    let candidates = merge_policy.compute_merge_candidates(segments.as_slice());
+    for candidate in candidates {
+        futures.push(writer.merge(&candidate.0));
+    }
+
+    let mut handles = vec![];
+    for future in futures {
+        let handle = std::thread::spawn(move || future.wait().expect("merge future should succeed"));
+        handles.push(handle);
+    }
```

Comment:
I don't feel like we should do these in parallel.  It could completely swamp disk I/O, which is not nice.